packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -37,7 +37,26 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
         const { uid } = sessionToken;
         const { recoveryKeyId, recoveryData, enabled } = request.payload;
 
-        await db.createRecoveryKey(uid, recoveryKeyId, recoveryData, enabled);
+        // Users that already have an enabled recovery key can not
+        // create a second recovery key
+        try {
+          await db.createRecoveryKey(uid, recoveryKeyId, recoveryData, enabled);
+        } catch (err) {
+          if (err.errno !== errors.ERRNO.RECOVERY_KEY_EXISTS) {
```

Comment:
This use of try-catch for flow control seems like an antipattern. Is there an alternative we could use instead? For instance, if `db.createRecoveryKey` was changed to `db.upsertRecoveryKey`, where the step of deleting any half-baked key could be abstracted into the database, the code would be simpler: "if recovery key exists, bail, else, upsert recovery key".